docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+
+## Types of recommendations
+
+Each article presents four types of recommendations: **Do**, **Consider**, **Avoid**, and **Do not**. The type of recommendation indicates how strongly it should be followed.
+
+You should almost always follow a **Do** recommendation. For example:
+
+✔️ DO include a short description for your package that describes what it's for.
+
+On the other hand, **Consider** recommendations should generally be followed, but there are legitimate exceptions to the rule and you shouldn't feel bad about not following the guidance:
+
+✔️ CONSIDER choosing a NuGet package name with a prefix that meets NuGet's prefix reservation [criteria](https://docs.microsoft.com/en-us/nuget/reference/id-prefix-reservation).
+
+**Avoid** recommendations mention things that are generally not a good idea, but breaking the rule sometimes makes sense:
+
+❌ AVOID NuGet package references that demand an exact version.
+
+And finally, **Do not** recommendations indicate something you should almost never do:
+
+❌ DO NOT use the `LicenseUrl` metadataproperty.
+
+## Framework targeting
+
+Authors should consider making their packages as inclusive as possible if they would like to maximize the number of potential consumers and overall impact in the .NET ecosystem.
+
+Highly inclusive packages are cross-platform and compatible with as many frameworks as possible. Today, that means using .NET Standard. .NET Standard is a specification of .NET APIs that are available on all .NET implementations.
+
+✔️ DO start with including a `netstandard2.0` target.
+> Most general-purpose libraries should not need APIs outside of .NET Standard 2.0. .NET Standard 2.0 is supported by all modern platforms and is the recommended way to support multiple platforms with one target.
+
+See the .NET [Cross-platform targeting guidance](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting) for more advanced suggestions regarding cross-platform and multitargeting scenarios.
+
+## Package metadata
+
+Metadata is a foundational component of any NuGet package. The quality of your metadata can vastly influence the discoverability and usability of your package.
+
+In Visual Studio, the recommended way to specify package metadata is to go Project > [Project Name] Properties > Package.
+
+Package metadata elements can also be specified directly in the project file (.csproj).
+
+Below is a table mapping and describing all the available packages metadata elements:
+
+| Visual Studio property name | csproj/MSBuild property name | Nuspec property name        | Description                                                                                                       |
```

Comment:
Why is this mapping table here? It doesn't have much to do with best practices themselves, right? Wouldn't this be better off in its own document?

Reply:
[It already is](https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#pack-target), although in a place that may not be obvious to many people. But that's a separate issue. Since you pointed it out, I agree, it doesn't seem like it should be in the recommended practises document.

Reply:
I'm not sure I agree. 

We don't go into detail on how to specify all these properties in the [Create a package documentation](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package-dotnet-cli). Someone reading this shouldn't have to look up a reference doc to to implement the suggestions. 

The table @zivkan linked also doesn't include Visual Studio property fields in the mapping (which may be more useful to beginners). 

Toward the goal of making this document as actionable as possible, do you think there is a more useful way we could present this information?

Reply:
Documentation on how to modify package metadata is useful not only to a best practices document, but to all customers creating packages. Therefore, if the "visual studio property fields" is useful, it should be available somewhere under the "authoring packages" section.

About making this document as actionable as possible, my feedback is that my personal preference is that the document contains information relevant to the topic at hand, which is recommendations. If I'm intermediate to experienced with NuGet, reading a document that duplicates beginner information will make me lose interest quickly, and I might miss the primary information the documentation author intended for me to learn.  But I understand that this is not as beginner friendly. You're the PM, so I'll leave it up to you where you want the docs to end up.

Something else I'll mention is that the more information we duplicate, the higher the risk that the docs get outdated/wrong if something in the product changes and only some of the duplicated docs are changed.

Reply:
@zivkan brings up good points. Personally, I'd vote to link to the relevant metadata docs for each recommendation.

Reply:
@loic-sharma @JonDouglas thoughts on including the whole table vs. linking to it in another doc?

Reply:
My vote is to put the fields/metadata referred in this doc here and a link to the complete (?) set of metadata towards the end. The best practices guide would not mention all the fields available for authors  and as we make updates to it, we can remember to update it here.

Reply:
Documentation duplication scares me so much. 
It's very easy to end up in a situation where one thing is updated, while the other one isn't. 

Is there a way to integrate parts of other docs into this one? 
I know there was a way to do it with the code snippets that we did for the nuget-sdk samples.